### ISSUE TITLE
Implement sudden death mechanics

### ISSUE THREAD

**Issue by trundlr** [apetavern/sbox-grubs#275]
A game should have a time or turn limit. Once the limit is reached, the water should rise (or the terrain should get lowered). This will reduce the amount of land available to traverse, and keep things interesting.

**Comment by ShadowBrian** [comment]
I started on this but can't figure out a reliable way of doing round counting and it needs a bunch of UI work to convey what's going on, it's pushed to a branch: https://github.com/apetavern/sbox-grubs/tree/ffa-sudden-death

Todo:
- Make round counting consistent to only lower the terrain after everyone who is alive has had a turn.
- Add settings for it to the config that it adheres by, like amount of rounds or time it takes before it initiates sudden death.
- Implement some sort of countdown UI for it.
- Add the option to reduce grub health to 1 when it's triggered.

**Comment by DrewBritt** [comment]
I finished the rest of the todo list. My only problem left is that during testing, I've managed to glitch Grubs into the terrain. This happens mostly when they're in a cave (but any terrain close enough to the top of their head causes it) and it drops onto their head, causing them to phase inside.

The obvious fix would be to swap over to raising the water level instead. Raising the KillZone is trivial, however, raising the water will require a bit more work due to it being static. On the flip side, it may be worth attempting to fix whatever collision issues the terrain is having, allowing us to do more interesting live terrain transformations.
